lib/ansible/galaxy/collection/__init__.py

```diff
@@ -504,8 +504,20 @@ def build_collection(u_collection_path, u_output_path, force):
     return collection_output
 
 
+def process_virtual_requirements(requirement, resolved_requirements, artifacts_manager):
```

Comment:
This name is confusing to me:
1) It seems to only process one virtual requirement at a time (so why use plural?)
2) Saying “process” is too generic, it doesn't really suggest what the implication of such processing is?

Looking at the implementation, it seems like it attempts to extract non-virtual dependencies. Maybe we could rename it?

Although, I don't quite like that it mutates an incoming param instead of returning or yielding stuff. I'd say that yielding would be more suiting here.

Reply:
It's plural because it's finding all leaf nodes for an ephemeral collection. So it should end up with possibly multiple `type: dir` requirements.

And yeah, didn't think too long about the naming. If you have a preference, feel free to suggest a new name.

Reply:
Now it's in providers and called `get_virtual_requirement_dependencies`.